src/model.jl

```diff
@@ -332,18 +332,28 @@ function _add_constraints_demand_flexibility!(
     sets::Sets,
     demand_flexibility::DemandFlexibility,
     interval_length::Int,
+    init_shifted_demand::Array{Float64,1}=Float64[],
 )
     if demand_flexibility.rolling_balance
+        println("rolling load balance, first window: ", Dates.now())
+        JuMP.@constraint(
+            m,
+            rolling_load_balance_first[i in 1:sets.num_load_bus],
+            sum(
+                m[:load_shift_up][i, j] - m[:load_shift_dn][i, j] 
+                for j in 1:(demand_flexibility.duration - 1)
```

Comment:
Should `j` be ranged in `[1, demand_flexibility.duration]` here?

Reply:
I don't think so. The idea for incorporating the previous interval's net demand deviation to the first rolling balance window is that the net demand deviation is like time `j = 0`. So, if we have `demand_flexibility.duration = 6`, then the first window would include 0 (the previous interval's net demand deviation) and 1, 2, 3, 4, and 5 (`1:(demand_flexibility.duration - 1)`)

Reply:
Should it be interpreted as "demand deviation of the first duration window + leftovers from previous interval >= 0"? Here, `j` should loop through every hour in the first duration window, no?

Reply:
Yeah, that's pretty much what I was trying to go for. Except, the first duration window is one hour shorter than normal (`(demand_flexibility.duration - 1)`) because the "leftovers from previous interval" are essentially take up one time step

Reply:
I see. Is that the classic way to model this? It seems a little weird to me since in the very first duration, i.e. first duration at the first interval, this gives the load balancing among hours `demand_flexibility.duration - 1`. Also, we didn't follow the same logic in interval balance either, i.e. `init_shifted_demand` is added into the constraint over the entire interval instead of excluding the last hour of the interval?